store/postgres/src/store.rs

```diff
@@ -23,18 +24,134 @@ use graph::{tokio, tokio::timer::Interval};
 use graph_graphql::prelude::api_schema;
 
 use chain_head_listener::ChainHeadUpdateListener;
-use entity_changes::EntityChangeListener;
 use functions::{
-    attempt_chain_head_update, build_attribute_index, lookup_ancestor_block, revert_block,
-    set_config,
+    attempt_chain_head_update, build_attribute_index, lookup_ancestor_block, pg_notify,
+    revert_block, set_config,
 };
+use store_events::StoreEventListener;
 
 embed_migrations!("./migrations");
 
-/// Internal representation of a Store subscription.
-struct Subscription {
-    pub entities: Vec<SubgraphEntityPair>,
-    pub sender: Sender<EntityChange>,
+/// The kind of event that we get from Postgres: either a notification about
+/// a change to a specific entity, or that we are done with a certain block
+#[derive(Clone, Debug, Deserialize)]
+pub enum StoreEvent {
+    Change(EntityChange),
+    Tick(BlockTick),
+}
+
+/// Internal representation of a Store subscription. Calling `subscribe`
+/// results in us storing an `EntityChange` sender, calling `block_ticks`
+/// results in us storing a `BlockTick` sender
+#[derive(Clone, Debug)]
+enum Subscription {
+    // We have entities in both kinds of subscriptions; since we clone the
+    // entire subscription in a few places, we make the entities an Arc
+    // to make the cloning cheaper.
+    Changes {
+        entities: Arc<Vec<SubgraphEntityPair>>,
+        sender: Sender<EntityChange>,
+    },
+    Ticks {
+        entities: Arc<Vec<SubgraphEntityPair>>,
+        sender: Sender<BlockTick>,
+    },
+}
+
+/// Track which subscriptions have had changes to their blocks. The entries
+/// in the set are the id's of subscriptions for which we've seen an
+/// EntityChange matching their interesting entities. When the corresponding
+/// BlockTick comes by, we pass a BlockTick to the receiver and remove
+/// the id from the set
+#[derive(Clone)]
+struct DirtyBlocks {
```

Comment:
I find the name `DirtyBlocks` non-obvious because what it really represents are IDs of subscriptions that have had changes to their blocks. I know it's long but how about `SubscriptionsWithBlockChanges`?